src/Processors/Formats/LazyOutputFormat.cpp

```diff
@@ -19,38 +19,20 @@ Block LazyOutputFormat::getBlock(UInt64 milliseconds)
     if (!queue.tryPop(chunk, milliseconds))
         return {};
 
-    if (!chunk)
-        return {};
-
-    auto block = getPort(PortKind::Main).getHeader().cloneWithColumns(chunk.detachColumns());
-    info.update(block);
-
-    if (auto chunk_info = chunk.getChunkInfo())
-    {
-        if (const auto * agg_info = typeid_cast<const AggregatedChunkInfo *>(chunk_info.get()))
-        {
-            block.info.bucket_num = agg_info->bucket_num;
-            block.info.is_overflows = agg_info->is_overflows;
-        }
-    }
+    if (chunk)
+        info.update(chunk.getNumRows(), chunk.allocatedBytes());
 
-    return block;
+    return chunk;
 }
 
-Block LazyOutputFormat::getTotals()
+Chunk LazyOutputFormat::getTotals()
 {
-    if (!totals)
-        return {};
-
-    return getPort(PortKind::Totals).getHeader().cloneWithColumns(totals.detachColumns());
+    return std::move(totals);
 }
 
-Block LazyOutputFormat::getExtremes()
+Chunk LazyOutputFormat::getExtremes()
```

Comment:
Am I right that 'totals' are number of rows we have read? And 'extemes' are minumums and maximums? And both of them are computed for every column?

Reply:
Yes, almost correct.
Extremes are min and max for every column. This chunk has 2 rows.
Totals are extra single row which make sense after group by. This row store default group by keys, and grouped values for all keys together.